aspnetcore/blazor/file-uploads.md

```diff
@@ -298,6 +335,34 @@ The following `FileUpload2` component:
 
 :::moniker-end
 
+:::moniker range=">= aspnetcore-8.0 < aspnetcore-9.0"
+
+If the component limits file uploads to a single file at a time or if the component only adopts interactive client-side rendering (CSR, `InteractiveWebAssembly`), the component can avoid the use of the `LazyBrowserFileStream` and use a <xref:System.IO.Stream>. The following demonstrates the changes for the `FileUpload2` component:
+
+```diff
+- var stream = new LazyBrowserFileStream(file, maxFileSize);
+- var fileContent = new StreamContent(stream);
++ var fileContent = new StreamContent(file.OpenReadStream(maxFileSize));
+```
+
+Remove the `LazyBrowserFileStream` class (`LazyBrowserFileStream.cs`), as it isn't used.
+
+:::moniker-end
+
+:::moniker range="< aspnetcore-8.0"
+
+If the component limits file uploads to a single file at a time, the component can avoid the use of the `LazyBrowserFileStream` and use a <xref:System.IO.Stream>. The following demonstrates the changes for the `FileUpload2` component:
+
+```diff
+- var stream = new LazyBrowserFileStream(file, maxFileSize);
+- var fileContent = new StreamContent(stream);
++ var fileContent = new StreamContent(file.OpenReadStream(maxFileSize));
+```
+
+Remove the `LazyBrowserFileStream` class (`LazyBrowserFileStream.cs`), as it isn't used.
+
+:::moniker-end
```

Comment:
Would we consider making the default example not use `LazyBrowserFileStream`, but then add a note explaining that if you're using server interactivity and support multiple file upload, the `LazyBrowserFileStream` approach might be necessary (and we link to the troubleshooting section which has the sample)? If you disagree, feel free to keep it how it is 🙂

Reply:
> if you're using server interactivity and support multiple file upload

That's exactly the scenario of this section. I think we should get a ruling on it from @danroth27. Broken devs can get angry 😡 with a 🦖 when we give them code that 💥. Personally, I'd like to try and prevent it from happening ... and head off the doc issues 💸.

Reply:
Sounds good, we can keep it as is!